### PR TITLE
Added new "filters" python file, along with a "hash" filter. This can…

### PR DESCRIPTION
**Pull request by alexander-marquardt**
Added new Jinja filter for calculating the hash of fields in the low code connector/transformation/AddFields functionality. 

I added a new `filters.py` python file, in which I have declared a "hash" filter. This can be extended to include other custom filters in the future. I also modified `jinja.py` so that the new hash functionality is available from the low code yaml declaration. 

This can be used in the calculation of values in the `AddFields` transformation as follows:  `value: "{{ record['rates']['CAD'] | hash('md5', 'mysalt')  }}"`

Addresses : [Creation of a Jinja filter that allows hashing of sensitive data before arriving at a destinationn](https://github.com/airbytehq/airbyte/issues/17993)

## What
Adds a custom jinja filter for hashing within the low-code transform functionality. 

## How
Wrapper around hashlib

## 🚨 User Impact 🚨
Will allow additional functionality that users did not previously have, and will better support GDPR compliance. 

## Pre-merge Checklist
Not sure .. will need to discuss with engineering

### Community member or Airbyter
Airbyter

### CODE REVIEW

**Review by girarda — COMMENTED** [review]
Love it! Thanks for contributing @alexander-marquardt

**Comment by alexander-marquardt** [comment]
Note: the md5 hash, when executed without a salt is not matching the expected md5 hash , from command line, which can be validated as follows: ``` echo -n foobar |md5   ``` versus the jinja code: ```"{{ 'xxx' | hash('md5')  }}"```. 

I will figure out the reason for this difference and update the code once I understand the reason this is happening.

**Comment by alexander-marquardt** [comment]
Oh, that was caused the application of a "None" salt when it was not passed in ... this was fixed by placing a condition around the application of the salt so that it does not modify the hash unless it is explicitly used.

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "rates"
+      primary_key: "date"
+      path: "/exchangerates_data/latest"
+    transformations:
+      - type: AddFields
+        fields:
+          - path: ["some_new_path"]
+            value: "{{ record['rates']['CAD'] | hash('md5', 'mysalt')  }}"
```
[the remark this answers is not in the document]
Comment (alexander-marquardt) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py:25]: I'll add a unit test that verifies the functionality

**Comment by alexander-marquardt** [comment]
@girarda  - I have added unit tests to validate the hash functionality

**Comment by girarda** [comment]
@alexander-marquardt this looks good. Only outstanding issue is the flake check failing because of unused imports in test_filters .py

>  ./unit_tests/sources/declarative/interpolation/test_filters.py:7:1: F401 'pytest' imported but unused
See https://github.com/airbytehq/airbyte/actions/runs/3265231904/jobs/5367150554

Next step will be update the [changelog](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/CHANGELOG.md) and bump the version in [setup.py](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/setup.py#L18)

**Comment by alexander-marquardt** [comment]
@girarda - I've updated the changelog and setup to reflect the newest version 0.1.102

**Review by girarda — APPROVED** [review]

**Pull request merged**